instructor/ui/sketch_tool.js

```diff
@@ -156,3 +156,20 @@ function printText(x, y, value){
     text_obj.position.x -= text_obj.bounds.width / 2;
     return text_obj;
 }
+
+/* Translate data points to normal coordinate system */
+function getDataPoints(){
+    var correct_points = [];
+    _.each(data_points, function(point) {
+        correct_points.push([point[0] - X_ORIGIN , Y_ORIGIN - point[1]]);
+    });
+    return correct_points;
+}
+/*Converts normal points to Canvas's point of reference
+* where origin is at top-left */
+function convertToCanvas(data){
```

Comment:
It's generally good practice to return a copy rather than mutate the parameter argument. This makes the function and its arguments easier to reason about. Can you change this to return a copy of `data`, maybe using `_.map` below?